### PR TITLE
imapflow typings

### PR DESCRIPTION
**Pull request by jeffreyratton98**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,8 +28,9 @@
         "@definitelytyped/definitions-parser": "latest",
         "@definitelytyped/dtslint-runner": "latest",
         "@definitelytyped/utils": "latest",
-        "danger": "^10.1.1",
         "@octokit/rest": "^16.0.0",
+        "@types/node": "^16.3.1",
+        "danger": "^10.1.1",
```
Comment (andrewbranch) [inline package.json:33]: Please revert package.json changes

**Review by andrewbranch — CHANGES_REQUESTED** [review]

**Comment by andrewbranch** [comment]
Hang on, imapflow [generates its own types from JSDoc](https://unpkg.com/browse/imapflow@1.0.59/lib/types.d.ts), it just doesn’t list them in its package.json so they likely won’t get pulled in... were you aware of this? Maybe you should work with the library author to improve that instead of duplicating them here. Alternatively, if the author wants to move them to DT, that’s fine, but they should delete the included typings if that’s the case. Can you try to sort this out with the library author? We would like to avoid having duplicated/conflicting definitions here. Thanks!

**Comment by jeffreyratton98** [comment]
> Hang on, imapflow [generates its own types from JSDoc](https://unpkg.com/browse/imapflow@1.0.59/lib/types.d.ts), it just doesn’t list them in its package.json so they likely won’t get pulled in... were you aware of this? Maybe you should work with the library author to improve that instead of duplicating them here. Alternatively, if the author wants to move them to DT, that’s fine, but they should delete the included typings if that’s the case. Can you try to sort this out with the library author? We would like to avoid having duplicated/conflicting definitions here. Thanks!

From the most recent issue regarding types:
According to the author, he would like to see the types moved to DT.  See his comment here:
https://github.com/andris9/imapflow/issues/4#issuecomment-704312044

Since his generated types are not linked in the package.json, we shouldn't have any issues with collisions, correct?

**Comment by andrewbranch** [comment]
Sounds good, just wanted to make sure everyone was on the same page here.

**Review by andrewbranch — APPROVED** [review]

**Comment by jeffreyratton98** [comment]
Ready to merge

**Pull request merged**